src/DynamoCore/Logging/AnalyticsService.cs

```diff
@@ -9,41 +9,34 @@ namespace Dynamo.Logging
     /// <summary>
     /// Utility class to support analytics tracking.
     /// </summary>
-    class AnalyticsService
+    internal class AnalyticsService
     {
         // Use the Analytics.Core interface so that we do not have to load the ADP assembly at this time.
         private static IAnalyticsUI adpAnalyticsUI;
 
         /// <summary>
-        /// Starts the client when DynamoModel is created. This method initializes
+        /// Indicates that we don't want to shut down analytics when DynamoModel shuts down.
+        /// Sometimes we want to keep Analytics service running even when we don't have a DynamoModel started.
+        /// </summary>
+        internal static bool KeepAlive { get; set; }
+
+        /// <summary>
+        /// Starts the Analytics client. This method initializes
         /// the Analytics service and application life cycle start is tracked.
         /// </summary>
-        /// <param name="model">DynamoModel</param>
-        /// <param name="isHeadless">Analytics won't be started if IsHeadless, but ADP may be loaded to be disabled.</param>
-        /// <param name="isTestMode">Analytics won't be started if isTestMode, ADP will not be loaded.</param>
-        internal static void Start(DynamoModel model, bool isHeadless, bool isTestMode)
+        internal static void Start(DynamoModel model)
         {
-            if (isTestMode)
-            {
-                if (Analytics.DisableAnalytics)
-                {
-                    model.Logger.Log("Incompatible configuration: [IsTestMode] and [Analytics disabled] ");
-                }
-                return;
-            }
-
-            if (isHeadless)
-            {
-                return;
-            }
-
             // Initialize the concrete class only when we initialize the Service.
             // This will also load the Analytics.Net.ADP assembly
             // We must initialize the ADPAnalyticsUI instance before the Analytics.Start call.
             adpAnalyticsUI = new ADPAnalyticsUI();
 
-            Analytics.Start(new DynamoAnalyticsClient(model));
-            model.WorkspaceAdded += OnWorkspaceAdded;
+            Analytics.Start(new DynamoAnalyticsClient(DynamoModel.HostAnalyticsInfo));
+
+            if(model != null)
+            {
+                model.WorkspaceAdded += OnWorkspaceAdded;
```

Comment:
We may need to unsubscribe it somewhere, looks like this was missing before

Reply:
Yes, and actually I want to piggy back on the original idea here and not have the AnalyticsService linked to DynamoModel - I think we can do this easier by simply calling into analytics when the workspace changes ( pretty much like we do with all the other tracking ) instead of subscribing and unsubscribing to model events.

Reply:
Thanks!